src/index.ts

```diff
@@ -56,7 +56,7 @@ export const createAuthProvider = <T>({
             return () => {
                 tp.unsubscribe(listener);
             };
-        }, [listener]);
+        });
```

Comment:
Am I right that it will unsubscribe/subscribe on each render? Maybe it's better to wrap `listener` with `useCallback`?

Smth like:
```
const useAuth = () => {
        const [isLogged, setIsLogged] = useState(tp.isLoggedIn());

        const  listener = useCallback((newIsLogged: boolean) => {
            setIsLogged(newIsLogged);
        }, [setIsLogged]);

        useEffect(() => {
            tp.subscribe(listener);
            return () => {
                tp.unsubscribe(listener);
            };
        }, [listener]);

        return [isLogged] as [typeof isLogged];
    };
```
(I didn't check the code)

Or it will be the same?

Reply:
один фиг

Reply:
я руководствовался [примером из документации реакт](https://ru.reactjs.org/docs/hooks-overview.html)
```js
import React, { useState, useEffect } from 'react';

function useFriendStatus(friendID) {
  const [isOnline, setIsOnline] = useState(null);

  function handleStatusChange(status) {
    setIsOnline(status.isOnline);
  }

  useEffect(() => {
    ChatAPI.subscribeToFriendStatus(friendID, handleStatusChange);
    return () => {
      ChatAPI.unsubscribeFromFriendStatus(friendID, handleStatusChange);
    };
  });

  return isOnline;
}
```